From 0be1c5728173ea9ac42843058e26b6268568acf0 Mon Sep 17 00:00:00 2001
From: Jiawen Geng <technicalcute@gmail.com>
Date: Fri, 14 Oct 2022 09:54:33 +0800
Subject: [PATCH] deps: V8: cherry-pick c2792e58035f
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Original commit message:

    [base] Fix build with gcc-13

    See https://gcc.gnu.org/gcc-13/porting_to.html#header-dep-changes.

    Also see Gentoo Linux bug report: https://bugs.gentoo.org/865981

    Change-Id: I421f396b02ba37e12ee70048ee33e034f8113566
    Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3934140
    Reviewed-by: Clemens Backes <clemensb@chromium.org>
    Reviewed-by: Simon Zünd <szuend@chromium.org>
    Commit-Queue: Clemens Backes <clemensb@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#83587}

Refs: https://github.com/v8/v8/commit/c2792e58035fcbaa16d0cb70998852fbeb5df4cc
PR-URL: https://github.com/nodejs/node/pull/44961
Fixes: https://github.com/nodejs/node/issues/43642
Reviewed-By: Michaël Zasso <targos@protonmail.com>
Reviewed-By: Richard Lau <rlau@redhat.com>
Reviewed-By: Luigi Pinca <luigipinca@gmail.com>
Reviewed-By: Colin Ihrig <cjihrig@gmail.com>

diff --git a/common.gypi b/common.gypi
index ccfd1bccb1..00f545ba6a 100644
--- a/common.gypi
+++ b/common.gypi
@@ -36,7 +36,7 @@
 
     # Reset this number to 0 on major V8 upgrades.
     # Increment by one for each non-official patch applied to deps/v8.
-    'v8_embedder_string': '-node.12',
+    'v8_embedder_string': '-node.16',
 
     ##### V8 defaults for Node.js #####
 
diff --git a/deps/v8/AUTHORS b/deps/v8/AUTHORS
index fd10aa4e36..3ef613e0f3 100644
--- a/deps/v8/AUTHORS
+++ b/deps/v8/AUTHORS
@@ -255,6 +255,7 @@ Vlad Burlik <vladbph@gmail.com>
 Vladimir Krivosheev <develar@gmail.com>
 Vladimir Shutoff <vovan@shutoff.ru>
 Wael Almattar <waelsy123@gmail.com>
+WANG Xuerui <git@xen0n.name>
 Wei Wu <lazyparser@gmail.com>
 Wenlu Wang <kingwenlu@gmail.com>
 Wenming Yang <yangwenming@bytedance.com>
diff --git a/deps/v8/src/base/logging.h b/deps/v8/src/base/logging.h
index 08db24a947..38be165fc1 100644
--- a/deps/v8/src/base/logging.h
+++ b/deps/v8/src/base/logging.h
@@ -5,6 +5,7 @@
 #ifndef V8_BASE_LOGGING_H_
 #define V8_BASE_LOGGING_H_
 
+#include <cstdint>
 #include <cstring>
 #include <sstream>
 #include <string>
diff --git a/deps/v8/src/inspector/v8-string-conversions.h b/deps/v8/src/inspector/v8-string-conversions.h
index c1d69c18f0..eb33c6816a 100644
--- a/deps/v8/src/inspector/v8-string-conversions.h
+++ b/deps/v8/src/inspector/v8-string-conversions.h
@@ -5,6 +5,7 @@
 #ifndef V8_INSPECTOR_V8_STRING_CONVERSIONS_H_
 #define V8_INSPECTOR_V8_STRING_CONVERSIONS_H_
 
+#include <cstdint>
 #include <string>
 
 // Conversion routines between UT8 and UTF16, used by string-16.{h,cc}. You may
-- 
2.20.1

